include/shell/shell.h

```diff
@@ -1010,6 +1011,56 @@ int shell_getopt(const struct shell *shell, int argc, char *const argv[],
  * @return		Pointer to struct getopt_state.
  */
 struct getopt_state *shell_getopt_state_get(const struct shell *shell);
+
+#if defined CONFIG_SHELL_GETOPT_LONG
```

Comment:
API shall not be ifdefed: https://docs.zephyrproject.org/latest/guides/design_guidelines.html#conditional-data-and-apis
`Function declarations that are available only when the option is enabled should be provided unconditionally`